packages/framework/src/Model/Category/Category.php

```diff
@@ -210,11 +214,114 @@ protected function setTranslations(CategoryData $categoryData)
         }
     }
 
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoTitle($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->getSeoTitle();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoH1($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->getSeoTitle();
+    }
+
+    /**
+     * @param int $domainId
+     * @return bool
+     */
+    public function isEnabled($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->isEnabled();
+    }
+
+    /**
+     * @param $domainId
+     * @return bool
+     */
+    public function isVisible($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->isVisible();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getSeoMetaDescription($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->getSeoMetaDescription();
+    }
+
+    /**
+     * @param int $domainId
+     * @return string|null
+     */
+    public function getDescription($domainId)
+    {
+        return $this->getCategoryDomain($domainId)->getDescription();
+    }
+
+    /**
+     * @return \Doctrine\Common\Collections\ArrayCollection|\Shopsys\FrameworkBundle\Model\Category\CategoryDomain[]
+     */
+    public function getCategoryDomains()
+    {
+        return $this->domains;
+    }
+
     /**
      * @return \Shopsys\FrameworkBundle\Model\Category\CategoryTranslation
      */
     protected function createTranslation()
     {
         return new CategoryTranslation();
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $categoryData
+     */
+    private function setDomains(CategoryData $categoryData)
+    {
+        $domainIds = $this->getDomainIdsForCategoryDomainData($categoryData);
+
+        foreach ($domainIds as $domainId) {
+            if ($domainId !== 0) {
+                try {
+                    $categoryDomain = $this->getCategoryDomain($domainId);
+                } catch (CategoryDomainNotFoundException $e) {
+                    $categoryDomain = new CategoryDomain($this, $domainId);
+                }
+                $categoryDomain->setSeoTitle($categoryData->seoTitles[$domainId]);
+                $categoryDomain->setSeoH1($categoryData->seoH1s[$domainId]);
+                $categoryDomain->setSeoMetaDescription($categoryData->seoMetaDescriptions[$domainId]);
+                $categoryDomain->setDescription($categoryData->descriptions[$domainId]);
+                $categoryDomain->setEnabled($categoryData->enabled[$domainId]);
+                $this->domains[$domainId] = $categoryDomain;
+            }
+        }
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\CategoryData $categoryData
+     * @return array
+     */
+    private function getDomainIdsForCategoryDomainData(CategoryData $categoryData)
```

Comment:
Why not use just the `array_keys($categoryData->seoTitles)` as in Brand and Product? Provided data object should be valid - should have attributes set for every domain ID (for which the data object factory is responsible).